app/code/Magento/CacheInvalidate/Model/PurgeCache.php

```diff
@@ -131,10 +132,25 @@ private function sendPurgeRequestToServers($socketAdapter, $servers, $formattedT
                 $socketAdapter->read();
                 $socketAdapter->close();
             } catch (\Exception $e) {
-                $this->logger->critical($e->getMessage(), compact('server', 'formattedTagsChunk'));
-                return false;
+                $unresponsiveServerError[] = "Cache host: " . $server->getHost() . ":" . $server->getPort() .
+                    "resulted in error message: " . $e->getMessage();
+                continue;
```

Comment:
Unnecessary operator, that's the end of foreach anyway

Reply:
Removed and added a return false when no servers could be purged

Reply:
I don't see these changes in the pull request

Reply:
It is in: https://github.com/magento/magento2/pull/18852/commits/7f2c079e2ebbc604056ef359c4c627b9cd7c776d